sdks/python/apache_beam/runners/interactive/interactive_beam.py

```diff
@@ -539,8 +541,16 @@ def collect(pcoll, n='inf', duration='inf', include_window_info=False):
     # Run the pipeline and bring the PCollection into memory as a Dataframe.
     in_memory_square = head(square, n=5)
   """
+  # Remember the element type so we can make an informed decision on how to
+  # collect the result in elements_to_df.
   if isinstance(pcoll, DeferredBase):
-    pcoll = to_pcollection(pcoll)
+    # Get the proxy so we can get the output shape of the DataFrame.
+    element_type = pcoll._expr.proxy()
```

Comment:
It's interesting to use the proxy as the element type. Note I filed BEAM-11064 for creating a pandas typehint. Once that exists we should use it here, but this makes sense for now.

Reply:
Done, added a TODO associated with that JIRA to track this.